bigbone/src/test/kotlin/social/bigbone/api/method/AnnouncementMethodsTest.kt

```diff
@@ -0,0 +1,101 @@
+package social.bigbone.api.method
+
+import io.mockk.verify
+import org.amshove.kluent.AnyException
+import org.amshove.kluent.invoking
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldNotThrow
+import org.amshove.kluent.shouldThrow
+import org.junit.jupiter.api.Test
+import social.bigbone.MastodonClient
+import social.bigbone.Parameters
+import social.bigbone.api.exception.BigBoneRequestException
+import social.bigbone.testtool.MockClient
+
+class AnnouncementMethodsTest {
+    @Test
+    fun `Get all read announcements`() {
+        val client = MockClient.mock("announcements.json")
+        val announcementMethods = AnnouncementMethods(client)
+        val announcements = announcementMethods.getAllAnnouncements(withDismissed = true).execute()
+
+        announcements.size shouldBeEqualTo 3
+
+        verify {
+            client.get(
+                path = "/api/v1/announcements",
+                query = any<Parameters>()
+            )
+        }
+    }
+
+    @Test
+    fun `Dismiss an announcement and verify that the method was called correctly`() {
+        val client = MockClient.mock("announcements.json")
```

Comment:
Dismiss, add, and remove methods always return `{}` in their JSON response. I think we should reflect that here. You can add a new file `announcement_dismiss_success.json` with content `{}` and use that here, even if we don’t do anything with the content for now.

Reply:
I have made the changes, but I am not really sure what exactly we are doing here. So in this piece of code:
```
fun `Dismiss an announcement and verify that the method was called correctly`() {
    val client = MockClient.mock("announcement_dismiss_success.json")
    val announcementMethods = AnnouncementMethods(client)

    announcementMethods.dismissAnnouncement("8")

    verify {
        client.performAction(
            endpoint = "/api/v1/announcements/8/dismiss",
            method = MastodonClient.Method.POST
        )
    }
}
```
Just to make sure I understand correctly, 
- We mock the `announcments_dismiss_success.json` file which contains an empty list
- Make the announcementmethods object that takes the client as a part of its constructor
- Dismisses the announcement with announcement id 8
But how would it be able to dismiss the announcement with announcement id 8 when `announcement_dismiss_sucess.json` is empty? Shouldn't it return an error?
- The verify step is clear to me, it just checks whether the client actually called the perform action method with the correct arguements or not

Reply:
> But how would it be able to dismiss the announcement with announcement id 8 when announcement_dismiss_sucess.json is empty? Shouldn't it return an error?

Again: The `MockClient` merely returns the content of the JSON file you supply as a successful response. There’s no actual handling of the calls we send to it.

You can check how we set up the `MockClient`. Maybe it makes a bit more sense then. 😊 

For now, as you said, the only thing we really test here is that calling `dismissAnnouncement` then calls the `performAction` function on the `client` with the expected parameters.
We might want to improve that in the future and use a proper OKHTTP MockWebServer. But that’s not currently the case.